Source/RunActivity/Viewer3D/RollingStock/MSTSLocomotiveViewer.cs

```diff
@@ -2564,7 +2564,16 @@ public ThreeDimentionCabViewer(Viewer viewer, MSTSLocomotive car, MSTSLocomotive
                                 break;
                             default:
                                 //cvf file has no external wipers, left door, right door and mirrors key word
-                                style = locoViewer.ThreeDimentionCabRenderer.ControlMap[key];
+                                if (locoViewer.ThreeDimentionCabRenderer.ControlMap.ContainsKey(key))
+                                {
+                                    style = locoViewer.ThreeDimentionCabRenderer.ControlMap[key];
+                                }
+                                else
+                                {
+                                    var cvfBasePath = Path.Combine(Path.GetDirectoryName(Locomotive.WagFilePath), "CABVIEW");
+                                    var cvfFilePath = Path.Combine(cvfBasePath, Locomotive.CVFFileName);
+                                    Trace.TraceWarning("Cabview control " + tmp[0].Trim() + " has not been defined in CVF file " + cvfFilePath);
+                                }
```

Comment:
```suggestion
                                if (!locoViewer.ThreeDimentionCabRenderer.ControlMap.TryGetValue(key, out style))
                                {
                                    var cvfBasePath = Path.Combine(Path.GetDirectoryName(Locomotive.WagFilePath), "CABVIEW3D");
                                    var cvfFilePath = Path.Combine(cvfBasePath, Locomotive.CVFFileName);
                                    Trace.TraceWarning($"Cabview control {tmp[0].Trim()} has not been defined in CVF file {cvfFilePath}");
                                }
```

Reply:
The 3D cab is using the same CVF as the 2D cab, and this one is normally in CABVIEW.
When I edited the CVF file in the CABVIEW directory, it has made a change on the 3D cab too.